textures/metro_control_panel_001_pegboard.py

```diff
@@ -25,10 +32,95 @@ def _remove_empty(pegs: Pegs) -> Pegs:
 
 
 def _swap_pegs(pegs: Pegs, a: str, b: str) -> Pegs:
+    """Trade places of two symbols"""
     assert len(a) == 1 and len(b) == 1
 
     placeholder = "_"
     # noinspection PyTypeChecker
     return tuple(
         tuple(s.replace(a, placeholder).replace(b, a).replace(placeholder, b) for s in panel) for panel in pegs
     )
+
+
+def markers_to_bools(peg_str: str) -> Tuple[bool]:
+    """
+    The positions of the markers represented as booleans
+
+    :param peg_str: String of peg symbols
+    :return: List of booleans, true where each marker (M) was
+
+    >>> markers_to_bools("PME")
+    [False, True, False]
+    """
+    return tuple(peg == "M" for peg in peg_str)
+
+
+def pegs_to_bools(peg_str: str) -> Tuple[bool]:
```

Comment:
these two functions basically do the same. Maybe use something like that:

```
def sequence_elements_equal(sequence: Sequence[T], needle: T) -> Tuple[bool]:
    return tuple(element == needle for element in sequence)
```

(just a blind guess, haven't tested it)

Reply:
Good call!